atat/src/traits.rs

```diff
@@ -180,37 +185,24 @@ pub trait AtatClient {
     fn reset(&mut self);
 }
 
-impl<T, L> AtatResp for heapless::Vec<T, L>
-where
-    T: AtatResp,
-    L: ArrayLength<T>,
-{
-}
-
-impl<L> AtatResp for heapless::String<L> where L: ArrayLength<u8> {}
+impl<T, const L: usize> AtatResp for Vec<T, L> where T: AtatResp {}
 
-impl<L> AtatCmd for heapless::String<L>
-where
-    L: ArrayLength<u8>,
-{
-    type CommandLen = L;
+impl<const L: usize> AtatResp for String<L> {}
 
-    type Response = heapless::String<heapless::consts::U256>;
+impl<const L: usize> AtatCmd<L> for String<L> {
+    type Response = String<256>;
     type Error = GenericError;
 
-    fn as_bytes(&self) -> Vec<u8, Self::CommandLen> {
+    fn as_bytes(&self) -> Vec<u8, L> {
         self.clone().into_bytes()
     }
 
     fn parse(
         &self,
         resp: Result<&[u8], &InternalError>,
     ) -> Result<Self::Response, Error<Self::Error>> {
-        resp.map(|r| {
-            heapless::String::from_utf8(Vec::from_slice(r).map_err(|_| Error::Parse)?)
-                .map_err(|_| Error::Parse)
-        })
-        .map_err(Error::from)?
+        let str = core::str::from_utf8(resp.map_err(Error::from)?).map_err(|_| Error::Parse)?;
+        Ok(String::from(str))
```

Comment:
`str` as variable name is confusing (even confuses GitHub's syntax highlighting).